### PR TITLE
Onnx tests

### PR DESCRIPTION
**Pull request by kjulian3**
First step in setting up a testing framework for python parsers to address #244. This pr adds a simple ONNX test case and adds the test to travis.

### CODE REVIEW

**Comment by kjulian3** [comment]
I've added tests to travis using the pytest package, and the tests run for all three build types. Since the language is cpp, I had to manually add python packages pip and setuptools, and update those packages. The windows build puts the compiled MarabouCore into the Release subfolder, so that has to be copied up a level to maraboupy to work correctly. The pr shows one example test case for a fully-connected, ReLU-activated network defined with ONNX. More cases can be added by adding additional functions in test_onnx.py, and additional files in the maraboupy/test folder will be run as well, so we can write a similar test_tf.py as well. Future PRs will add more tests and expand testing to tensorflow.

**Comment by kjulian3** [comment]
Lastly, pytest can also provide a simple code coverage report via pytest-cov. If we set it up with something like coveralls, we can get some nice interactive reports on what lines of the code have been tested, and which have yet to be tested. It would look something like this: https://coveralls.io/github/sisl/NNet

Are we interested in setting up coveralls, or some other form of code coverage?

**Comment by yuvaljacoby** [comment]
Two small comments:
* Did you try to work with travis cache ? it can cache a folder between iterations, then we will not need to download python every time (I see it takes only ~30 seconds and only on windows so it is not a must). 
* Do we want to run the test also as part of the build process (only if compiling with python support of course), in this way if the python API breaks we will know right away? 

regarding code coverage, I guess that we will want something that will support also c++ (I am not familiar with the tools you suggested). but anyway, sounds like a different task, so we can finish this one and then do that

**Comment by kjulian3** [comment]
I haven't worked with travis cache before. Is there a way to cache only for windows? Also I found online that we might be able to cache the python packages installed with pip using "cache: pip", but they say that language: python must be set. I'm not sure if it works with language: cpp.

I've add the tests to cmake along with a flag to turn on or off the tests. I updated the test file so the tests can be run from any directory.

File: CMakeLists.txt [hunk 1]
```diff
@@ -268,6 +269,22 @@ if (NOT ${TESTS_TO_RUN} STREQUAL "")
...
+        )
+    endif()
+
+    add_custom_command(
+        TARGET build-tests
+        POST_BUILD
+        COMMAND ${PYTHON_EXECUTABLE} -m pytest ${PYTHON_API_DIR}/test
+    )
+endif()    
+
```
Comment (yuvaljacoby) [inline CMakeLists.txt:287]: Did you test this on windows and unix?  shouldn't the second command be in else statement?

Comment (kjulian3) [reply CMakeLists.txt:287]: The second command runs the python api tests, which should be done for both windows and unix. Windows puts the MarabouCore shared library in a Release subdirectory, which needs to be copied to the maraboupy directory to be used. Once it's copied, the tests run the same way for all build types. I tested on my laptop, and travis was able to build correctly. The travis outputs show that the tests were run and passed after the build.

**Comment by yuvaljacoby** [comment]
Sorry for the late response.

I don't think we can cache only for windows (you can try to put an if statement there) 
But maybe we can cache an empty directory in unix? 

I will be able to take a deeper look only in about a week in a half

File: CMakeLists.txt (hunk 1 above)
Comment (yuvaljacoby) [reply CMakeLists.txt:287]: oops

**Comment by kjulian3** [comment]
I'll merge for now so I can start adding more test cases. We can look into adding to the travis cache in a future pr.

**Pull request merged**